### PR TITLE
[hotfix] Add cookie banner to legacy OSF frontend [PLAT-902]

### PR DESCRIPTION
**Pull request by erinspace**
## Purpose

Add a banner to alert non-logged in users that the OSF uses cookies. This is dismissible and shouldn't show up for logged-in users.

cookie-banner

Note: Appears over the slide-in banner that asks non-logged in users to join the OSF.

screen shot 2018-07-09 at 11 45 22 am

## Changes
- add HTML and CSS for banner
- javascript that adds a cookie when accepted and hides the banner

## QA Notes

This banner:
- Is for legacy OSF frontend
- Should only appear for non-logged-in users who haven't previously clicked Accept
- Clicking Accept sets a cookie that will hide the banner as long as that cookie is set
- The cookie banner should appear over the pre-existing "Join the OSF" banner that slides in
	- both alerts are separately dismissible.

## Documentation

None necessary

## Side Effects

None anticipated

## Ticket
https://openscience.atlassian.net/browse/PLAT-902

### CODE REVIEW

**Comment by caseyrollins** [comment]
@erinspace, the background color of the banner is meant to be the same color as the navbar. Are those the same colors in the screenshot and my eyes are just deceiving me?

**Comment by erinspace** [comment]
@caseyrollins I struggled with this too! According to the [OSF Style Guide](https://centerforopenscience.github.io/osf-style/style.html#colors) #204762 is indeed "OSF Blue" but I thought it also looked really light.  Doing a hover-over it looked more like #3333 but it's even a different color in [Baylee's PR](https://github.com/CenterForOpenScience/ember-osf/pull/397/files#diff-95d0a2ddc7950464fc6c5cec67aff318) sooo basically I'll do whatever, was just defaulting to the style guide!

**Comment by caseyrollins** [comment]
Ah, thank you for digging! It looks like #263947 to me when I inspect the element, but I can't seem to find that in any of the ember repositories. @laurenbarker or @brianjgeiger, can you point us to where the navbar color is defined?

**Comment by laurenbarker** [comment]
I believe it's [here](https://github.com/CenterForOpenScience/osf-style/blob/370b544affcdafc5734cdd388587575895e4b33a/css/base.css#L584) and [here](https://github.com/CenterForOpenScience/osf-style/blob/370b544affcdafc5734cdd388587575895e4b33a/css/base.css#L550). The navbar isn't actually a solid color :/

**Comment by erinspace** [comment]
@caseyrollins updated to a :hot: :fox: and changed the background color to [match the color in the osf-style repo](https://github.com/CenterForOpenScience/osf-style/blob/370b544affcdafc5734cdd388587575895e4b33a/css/base.css#L584)

screen shot 2018-07-10 at 1 15 10 pm

**Comment by erinspace** [comment]
Updated links to white underlined text as specified in https://github.com/CenterForOpenScience/ember-osf/pull/397#discussion_r201705581

screen shot 2018-07-11 at 11 21 01 am

**Comment by erinspace** [comment]
Updated link color!

screen shot 2018-07-12 at 1 57 49 pm

File: website/static/js/pages/base-page.js [hunk 1]
```diff
@@ -39,6 +39,24 @@ if (String.prototype.endsWith === undefined) {
...
 
+// Cookie banner notice for logged out users
+var cookieBannerSelector = '#cookieBanner';
+var CookieBannerViewModel = function(){
+    var self = this;
+    self.elem = $(cookieBannerSelector);
+    var cookieConsentKey = 'osf_cookieconsent';
+
+    self.accept = function() {
+        Cookie.set(cookieConsentKey, '0', { expires: 30, path: '/'});
```
Comment (caseyrollins) [inline website/static/js/pages/base-page.js]: The cookie value should be set to 1 (not 0).

File: website/static/js/pages/base-page.js [hunk 2]
```diff
@@ -39,6 +39,24 @@ if (String.prototype.endsWith === undefined) {
...
+var CookieBannerViewModel = function(){
+    var self = this;
+    self.elem = $(cookieBannerSelector);
+    var cookieConsentKey = 'osf_cookieconsent';
+
+    self.accept = function() {
+        Cookie.set(cookieConsentKey, '1', { expires: 30, path: '/'});
+    };
+
+    var accepted = Cookie.get(cookieConsentKey) === '0';
```
Comment (caseyrollins) [inline website/static/js/pages/base-page.js]: Need to check `=== '1'` here

**Pull request merged**